lms/views/admin/application_instance.py

```diff
@@ -264,12 +271,33 @@ def show_instance(self):
     def update_instance(self):
         ai = self._get_ai_or_404(**self.request.matchdict)
 
+        # Keep the same org if no public_id is sent
+        organization_id = ai.organization_id
+        if org_public_id := self.request.params.get("org_public_id", "").strip():
+            try:
+                if org := self.organization_service.get_by_public_id(org_public_id):
```

Comment:
The admin pages don't allow setting `None` for the organization_id.

I rather tackle that as a general update for all optional values.

Reply:
Should any of this checking and safety be in the service instead?

Reply:
Moved to the service